doc/src/overrides.md

```diff
@@ -97,6 +97,18 @@ just `nightly-2021-01-21`. The file has to be encoded in US-ASCII this case
 (if you are on Windows, check the encoding and that it does not start with a
 BOM). The legacy format is not available in `rust-toolchain.toml` files.
 
+If you ever get an error like the following (when running `rustc`, `cargo` or
+other command)
+
+```
+error: invalid channel name '[toolchain]' in '/PATH/TO/DIRECTORY/rust-toolchain'
+```
+
+it means you're running `rustup` pre-1.23.0 (or just a default toolchain set by
+`rustup` pre-1.23.0) and trying to interact with a project that uses the new TOML
+encoding in the `rust-toolchain` file.  You need to upgrade `rustup` to 1.23.0+
+and set the default toolchain again.
```

Comment:
There is no need to 'set the default toolchain again'.  You simply need to upgrade `rustup` (e.g. via `rustup self update` or `rustup update`)